### PR TITLE
DDF-3178 reduced the number of redundant memory copies that the tika input transformer was doing

### PR DESCRIPTION
**Pull request by stustison**
#### Who is reviewing it? 
(please choose AT LEAST two reviewers that need to approve the PR before it can get merged)
@clockard @spearskw  @brendan-hofmann @pklinef 
#### Select relevant component teams: 
https://github.com/orgs/codice/teams
#### Choose 2 committers to review/merge the PR. 
(please choose ONLY two committers from below, delete the rest)
@coyotesqrl
@jlcsmith
@stustison
#### What are the relevant tickets?
[](https://codice.atlassian.net/browse/)
#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests
#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: catalog/transformer/catalog-transformer-tika-input/src/main/java/ddf/catalog/transformer/input/tika/TikaInputTransformer.java [hunk 1]
```diff
@@ -639,17 +652,22 @@ private String transformToXml(String xhtml) {
...
+                            .openStream()) {
                 Transformer transformer = templates.newTransformer();
-                transformer.transform(new SAXSource(xmlReader,
-                        new InputSource(new StringReader(xhtml))), new StreamResult(xml));
-                return xml.toString();
-            } catch (TransformerException e) {
+                transformer.transform(new SAXSource(xmlReader, new InputSource(xhtmlInStream)),
+                        new StreamResult(xmlOutStream));
+                //we should not be doing this and should be returning the stream instead
+                try (InputStream resultSream = xmlOutStream.asByteSource()
```
Comment (coyotesqrl) [inline catalog/transformer/catalog-transformer-tika-input/src/main/java/ddf/catalog/transformer/input/tika/TikaInputTransformer.java]: ✏️ As little as we care, typo in `resultStream`

**Comment by stustison** [comment]
build now

**Comment by stustison** [comment]
build now

**Comment by stustison** [comment]
build now

**Pull request merged**